core/utilities.lua

```diff
@@ -103,9 +103,23 @@ utilities.debug = function (category, ...)
   end
 end
 
-utilities.dump = function (...)
+local _dumper = function(ret, space, ...)
```

Comment:
I don't think we want to complicated the `dump()` function much more. Also some versions of Lua have issues with this mixed variable argument format.

The way we've been doing this before is actually adding `__tostring()` methods to tables that need this kind of dumping, then using `SU.debug()`. Looking at your use case I think that's probably appropriate here too.

Reply:
Can @simoncozens please comment? I found this quite useful while debugging, and had it around in my files for a while. I can fix the mixed variable arguments. Having to add a metatable each time is annoying for what should be quick debugs, IMO. But, if @simoncozens hates it too, I'll remove it.

Reply:
Any table being used as a data type that's being contributed to SILE core should have meta methods that facilitate this sort of thing. Anything you're hacking on your end prior to contribution can use whatever methods you like (I have my own collection too) but contributing a debug function to SILE core that isn't the most appropriate way to debug stuff that's in SILE core doesn't feel right.

What data do you have that `SU.dump(data)` doesn't spit out for "quick debugs" anyway? or `SU.dump{data1, data2, data3}` if passing multiple data points for debugging is your issue. What's to gain by stripping line break info and dumping only string elements?